Source/Classes/PINRequestRetryStrategy.m

```diff
@@ -0,0 +1,69 @@
+//
+//  PINRequestRetryStrategy.m
+//  Pods
+//
+//  Created by Hovhannes Safaryan on 9/24/16.
+//
+//
+
+#import "PINRequestRetryStrategy.h"
+#import "PINURLSessionManager.h"
+#import "PINRemoteImageManager.h"
+
+@interface PINRequestExponentialRetryStrategy ()
+
+@property (nonatomic, assign) int retryMaxCount;
+@property (nonatomic, assign) int retryCount;
+@property (nonatomic, assign) int delayBase;
+
+@end
+
+@implementation PINRequestExponentialRetryStrategy
+
+- (instancetype)initWithRetryMaxCount:(int)retryMaxCount delayBase:(int)delayBase
+{
+    if (self = [super init]) {
+        _retryCount = 0;
+        _retryMaxCount = retryMaxCount;
+        _delayBase = delayBase;
+    }
+    return self;
+}
+
+- (int)numberOfRetries
+{
+    return self.retryCount;
+}
+
+- (BOOL)shouldRetryWithError:(NSError *)error
+{
+    if (error == nil || ![[self class] retriableError:error] ||
+        self.retryCount > self.retryMaxCount) {
```

Comment:
@hovox I think I get what you're saying, but for the first download attempt, before there is any retry, the retryCount is zero and is *not* incremented, correct? Then if the initial download fails, and it attempts to retry, it checks self.retryCount > self.retryMaxCount (3 lets say) and it succeeds. It does this 3 more times for a total of 4 retries when self.retryMaxCount is 3.

Or I'm missing something which is entirely possible :)